actionpack/lib/action_dispatch/middleware/stack.rb

```diff
@@ -130,7 +130,7 @@ def swap(target, *args, &block)
     ruby2_keywords(:swap)
 
     def delete(target)
-      middlewares.delete_if { |m| m.name == target.name }
+      middlewares.reject! { |m| m.name == target.name } || (raise "No such middleware to delete: #{target.inspect}")
```

Comment:
Keeping inline with `assert_index`, we could do something similar here:
```suggestion
      assert_deletable_target(target)
      middlewares.reject! { |m| m.name == target.name } 
```

Where assert_deletable_target could be something like:
```
def assert_deletable_target(target)
  raise "No such middleware to delete: #{target.inspect}" unless index_of(target)
end
```

Reply:
We could, but given it's one line either way I think it's readable enough as is.